client/src/components/AuthService/index.js

```diff
@@ -18,25 +18,34 @@ export default class AuthService {
       });
   };
 
+  // Get profile is called first
   getProfile = () => {
-    return decode(this.getToken());
+    const token = this.getToken();
+    if(token != null ){
+      console.log("The user prrovided a token")
+      return decode(token);
+    }else{
+      return false
```

Comment:
When it comes to this line, there is not much logic needed, because at this point the user does not have a token.